sonyc_kalman/data.py

```diff
@@ -70,6 +70,93 @@ def load_openl3_time_series(hdf5_path, delta_mins=15, aggr_func=None):
     return X, mask
 
 
+def series_splitter(invalid_mask, test_ratio=0.25, mode='random_holes', hole_param=[20, 10]):
+    '''
+    Given a `invalid_mask` of length n, generate masks for the train split and test split according to `test_ratio`.
+
+    Params:
+    -------
+        invalid_mask: list or np.array
+            an array of 1's and 0's. 0 being valid, 1 being invalid
+        test_ratio: float (0, 1)
+            the fraction of the data to be split out as test
+        mode: str
+            should be one of 'random_holes', 'r', 'chronological', or 'c'.
+            'random_holes' - punch_out randomly (Guasssian according to `hole_param`) sized holes in the series and 
+            set aside as test.
+            'chronological' - just use the begining as train and the end as test, modeling real world delopyment.
+            hole_param in chronological mode is ignored.
+        hole_param: list [mean, std]
+            used for the 'random_holes' mode.
+            
+    Returns:
+    --------
+        train_mask: np.array of int's
+            an array of 1's and 0's that can be used to mask any series. 1 means selected for the training split
+        test_mask: np.array of int's
+            an array of 1's and 0's that can be used to mask any series. 1 means selected for the test split
```

Comment:
Let's be consistent with our conventions. IIRC we decided on following numpy.ma conventions (1=masked, 0=readable), so we should probably switch these (unless I'm misinterpreting this). That way, we can just take our input data and apply a numpy mask to it to our data for training/evaluation. @jtcramer, let me know what you think of this.

Reply:
I'm in favor of consistency across the board.